scripts/pi-hole/php/savesettings.php

```diff
@@ -323,28 +347,35 @@ function addStaticDHCPLease($mac, $ip, $hostname) {
 					$extra .= "no-dnssec";
 				}
 
-				// Check if Conditional Forwarding is requested
-				if(isset($_POST["conditionalForwarding"]))
+				// Check if rev-server is requested
+				if(isset($_POST["rev_server"]))
 				{
-					$conditionalForwardingIP = trim($_POST["conditionalForwardingIP"]);
-					$conditionalForwardingDomain = trim($_POST["conditionalForwardingDomain"]);
+					// Validate CIDR IP
+					$cidr = trim($_POST["rev_server_cidr"]);
+					if (!validCIDRIP($cidr))
+					{
+						$error .= "Conditional forwarding subnet (\"".htmlspecialchars($cidr)."\") is invalid!<br>".
+						          "This field requires CIDR notation for local subnets (e.g., 192.168.0.0/16).<br>".
+						          "Please use only subnets /8, /16, /24, and /32.<br>";
```

Comment:
@DL6ER This works as expected, I get an error if I try to use a /23. But I do have a question, why is /23 invalid? 

My network spans 192.168.0.1-192.168.1.254

(I should have checked this before merging I guess!)

Reply:
https://discourse.pi-hole.net/t/support-prefix-12-for-conditonal-forwarding/33975?u=yubiuser

Reply:
Thanks.... I tried a cursory search on discourse to see if it had already been discussed. I failed, clearly!

Reply:
In any case, it works if I use /16

Reply:
I think there was a mailinglist patch or question to allow for non-natural masks.

Reply:
Yeah, for future reference:

http://lists.thekelleys.org.uk/pipermail/dnsmasq-discuss/2020q2/013985.html

http://lists.thekelleys.org.uk/pipermail/dnsmasq-discuss/2020q2/013997.html

Reply:
Yeah. TL;DR `dnsmasq` only supports multiples of 8 for IPv5 (8, 16, 24) and multiples of 4 for IPv6 (4, 8, 12, ...). I submitted patches to allow arbitrary prefix lengths but nothing happened to them. As odd prefix-lengths cause `dnsmasq` to fail on startup, they are forbidden for now (until my two patches are implemented)

Reply:
> IPv5

Reply:
I do this every now and then, usually it goes unnoticed :-)